spec/unit/policyfile_includes_spec.rb

```diff
@@ -0,0 +1,512 @@
+          expect { policyfile_lock.to_lock }.to raise_error(Solve::Errors::NoSolutionError) do |e|
+            expect(e.to_s).to match(/#{included_policy_lock_name}/)
+          end
+
+        end
+
+        it "includes the source of the conflicting dependency constraint from the including policy" do
+          expect { policyfile_lock.to_lock }.to raise_error(Solve::Errors::NoSolutionError) do |e|
+            expect(e.to_s).to match(/`cookbookC \(= 2.0.0\)`/) # This one comes from the included policy
+            expect(e.to_s).to match(/`cookbookC \(= 1.0.0\)` required by `local-1.0.0`/) # This one comes from the included policy
+          end
+        end
+
+        it "should not have an open constraint on cookbookC" do
+          pending("I think this is a bug")
+          expect { policyfile_lock.to_lock }.to raise_error(Solve::Errors::NoSolutionError) do |e|
+            expect(e.to_s).not_to match(/`cookbookC \(>= 0.0.0\)`/)
+          end
+        end
+
+      end
+
+      context "and the including policy explicitly sets the source for a cookbook which conflicts with the source in the included policy" do
+
+        it "raises an error describing the conflict"
+
+        it "includes the name and location of the included policy in the error message"
+
+      end
+
+    end
+
+    context "when default attributes are specified" do
+      let(:default_attributes) do
+        {
+          "shared" => {
+            "foo" => "bar",
+          },
+        }
+      end
+
+      context "when the included policy does not have attributes that conflict with the including policy" do
+        let(:included_policy_default_attributes) do
+          {
+            "not_shared" => {
+              "foo" => "bar",
+            },
+          }
+        end
+
+        it "emits a lockfile with the attributes from both merged" do
+          expect(policyfile_lock.to_lock["default_attributes"]).to include(included_policy_default_attributes, default_attributes)
+        end
+
+      end
+
+      context "when the included policy has attributes that conflict with the including policy, but provide the same value" do
+        let(:included_policy_default_attributes) { default_attributes }
+
+        it "emits a lockfile with the attributes from both merged" do
+          expect(policyfile_lock.to_lock["default_attributes"]).to eq(default_attributes)
+        end
+
+      end
+
+      context "when the included policy has attributes that conflict with the including policy's attributes" do
+        let(:included_policy_default_attributes) do
+          {
+            "shared" => {
+              "foo" => "not_bar",
+            },
+          }
+        end
+
+        it "raises an error describing all attribute conflicts" do
+          expect { policyfile_lock.to_lock }.to raise_error(ChefDK::Policyfile::AttributeMergeChecker::ConflictError)
+        end
+
+        it "includes the name and location of the included policy in the error message"
```

Comment:
Can probably add a test for the actual exception message to the test above and remove these examples.